src/AppInstallerCommonCore/Filesystem.cpp

```diff
@@ -64,10 +64,9 @@ namespace AppInstaller::Filesystem
 
     bool PathEscapesBaseDirectory(const std::filesystem::path& path, const std::filesystem::path& base)
     {
-        // Input for path must exist or this will throw a filesystem error.
-        const auto& pathString = std::filesystem::canonical(path).u8string();
-        const auto& baseString = std::filesystem::absolute(base).u8string();
-        return pathString.find(baseString) == std::string::npos;
+        const auto& canonicalPath = std::filesystem::weakly_canonical(path);
+        auto [a, b] = std::mismatch(canonicalPath.begin(), canonicalPath.end(), base.begin(), base.end());

```

Comment:
You do need to use `weakly_canonical` on the base path as well, or a symlink in the base path could result in a false negative result.

Reply:
Converted base path to weakly_canonical prior to comparison